### PR TITLE
Moves hbond_autocorrel and removes old hbond code

### LINKED ISSUE #2739: Deprecated and slated for 2.0 removal
We have quite a lot of things that are slated for removal in 2.0:

- [x] old density code #2740 
- [x] old hole code #2778 
- [ ] old hbonds code #2746 
- [x] deprecated writers Timestep code #2754 
- [x] progressmeter #2743 

With #2729 I guess we started 2.0 a little bit sooner than anticipated, so I thought I'd check that we are good at doing these removals before we go ahead and start cutting out code?

### LINKED ISSUE #3259: Docs for WaterBridgeAnalysis incorrectly formatted/do not show up
## Expected behavior ##

WaterBridgeAnalysis https://docs.mdanalysis.org/2.0.0-dev0/documentation_pages/analysis/wbridge_analysis.html shows up as its own entry.

## Actual behavior ##

Instead it shows up under https://docs.mdanalysis.org/2.0.0-dev0/documentation_pages/analysis_modules.html#hydrogen-bonding

## Code to reproduce the behavior ##

See links above.

This looks like an issue with the title formatting.

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) 2.0.0-dev docs

### PR DESCRIPTION
**Pull request by IAlibay**
Fixes #2739 
Fixes #3259 

Changes made in this Pull Request:
 - Removes old hbond code
 - Add warning for upcoming move of hbond_autocorrel to hydrogenbonds (for consistency).

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/analysis/hydrogenbonds/hbond_autocorrel.py [hunk 1]
```diff
@@ -0,0 +1,608 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
```
Comment (IAlibay) [inline package/MDAnalysis/analysis/hydrogenbonds/hbond_autocorrel.py:1]: This is going to re-write the commit history for this file isn't it?

**Comment by IAlibay** [comment]
I realise we usually create stubs so that folks can change workflows ahead of time, however the removal of history for hbond_autocorrel bothered me. Since all it means to users is that they have to change their import, can we just do this as a warned move without moving things ahead of time? (worst case I create a stub in hydrogenbonds that imports hbond.hbond_autocorrel, but that makes warnings messy)

**Comment by orbeckst** [comment]
At the moment the test_api tests fail because there's nothing in hydrogenbonds. I'll push a change with the moved file and stubs in the old place. If you don't like it, revert it.

**Comment by IAlibay** [comment]
@orbeckst please go ahead, I was going to try to complete this tomorrow, but any push along is appreciated :)

**Comment by orbeckst** [comment]
I am also fixing #3259 because I have to edit the docs in wbridge_analysis anyway.

**Review by orbeckst — APPROVED** [review]
I am ok with your and my changes... if you or anyone else could have a quick look then that would be good.

@IAlibay I am leaving it to you to merge/request changes.

File: testsuite/MDAnalysisTests/analysis/test_hydrogenbondautocorrel_deprecated.py [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+
+
+def test_donors_nobonds():
+    u = mda.Universe(XYZ_mini)
+
+    with pytest.raises(mda.NoDataError):
+        hbonds.find_hydrogen_donors(u.atoms)
+
+
+def test_find_hydrogen_donors_deprecation_warning(u_water):
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/analysis/test_hydrogenbondautocorrel_deprecated.py:315]: @orbeckst is there any point in having most of the hbond_autocorrel tests duplicated here? They are pretty expensive iirc, if you're happy with it, we can just move these two warning tests over to the other hbond_autocorrel tests?

**Comment by orbeckst** [comment]
Tests run in seconds and I wanted to make sure that the stubs properly process arguments (I wasn’t 100% sure if I could use the deprecate decorator on a class like that..). 

For 2.0.0 I’d leave the full tests (no harm) and then we can slim it down later. I didn’t quite know which ones to remove but still test basic functionality.

**Comment by IAlibay** [comment]
@orbeckst - ok that's fair. I'm happy with this (except from being sad that we will lose git history for `hbond_autocorrel`) so I'll squash-merge.

**Pull request merged**